tt_metal/impl/allocator/algorithms/allocator_algorithm.hpp

```diff
@@ -58,11 +58,16 @@ class Algorithm {
 
     virtual void dump_blocks(std::ofstream &out) const = 0;
 
+    virtual void shrink_size(DeviceAddr shrink_size, bool bottom_up=true) = 0;
+
+    virtual void reset_size() = 0;
```

Comment:
would we want to do further divisions after a sub device has already been cut from full device and then have reset be like restoring size from some stack

Reply:
Currently for sub device there won't be further divisions as we can only load one sub device manager/configuration at a time, so having the generic stack support isn't required. Ex load sub device config 1 -> shrink(x), and to load a new config 2 we need to reset, then shrink(y).
I don't think the plan for distributed allocator needs multiple divisions either @tt-asaigal?
We could file issues to track these generality improvements. We had previously talked about supporting memory pools, and these suggestions seem like it would apply there if we do move forward with supporting that.

Reply:
I don't think distributed allocator needs to worry about subdivisions. For the distributed case, we will use a single allocator config across a Mesh. This allocator can be setup however we like, and we can partition it based on the SubDevice config (the same config will be applied to each device in the Mesh).